backend/service/github/github.go

```diff
@@ -96,9 +96,22 @@ type File struct {
 	LastModifiedSHA  string
 }
 
+type FileEntry struct {
+	Name string
+	Type string
+}
```

Comment:
just curious, is a FileEntry diff from a File from github's POV?

Reply:
in terms of the endpoint flow, with this info returned, is the plan then to make another API call for the contents?

Reply:
I dont 'think' so? Other than a File normally having the contents fetched already it's really just a summary of what's in a directory.

Yeah, my thinking is that we may only want to know what files are available and then another call/calls can be made for the contents, versus fetching everything already which could greatly increase the response size with data that may not be needed,